src/net.cpp

```diff
@@ -1688,8 +1719,17 @@ void ThreadOpenConnections()
             break;
         }
 
-        if (addrConnect.IsValid())
-            OpenNetworkConnection(addrConnect, (int)setConnected.size() >= std::min(nMaxConnections - 1, 2), &grant);
+        if (addrConnect.IsValid()) {
+
+            if (fFeeler) {
+                // Add small amount of random noise before connection to avoid synchronization.
+                int randsleep = GetRandInt(FEELER_SLEEP_WINDOW * 1000);
```

Comment:
Randomizing the interval was suggested by @sipa and it is an excellent idea!  This pull request already does exactly what you describe using PoissonNextSend.